cpp/program/play.cpp

```diff
@@ -916,6 +916,15 @@ static void recordTreePositionsRec(
     SidePosition* sp = new SidePosition(board,hist,pla,numNeuralNetChangesSoFar);
     extractPolicyTarget(sp->policyTarget, toMoveBot, node, locsBuf, playSelectionValuesBuf);
     extractValueTargets(sp->whiteValueTargets, toMoveBot, node);
+
+    double policySurprise = 0.0, policyEntropy = 0.0, searchEntropy = 0.0;
+    bool success = toMoveBot->getPolicySurpriseAndEntropy(policySurprise, searchEntropy, policyEntropy);
```

Comment:
You mentioned out of band that the changes to the side position calculation here to support finding policy surprise and entropy should be more complicated, but I cannot see how.  I suspect that it has something to do with the state of the `Search` object that I'm calling against.  Can you clarify?

Reply:
The node passed in here to the other functions (like extractPolicyTarget) is not the root node of the search, whereas getPolicySurpriseAndEntropy doesn't take a node as an argument and only computes the value for the root node. That makes doing this correctly a bit messier. :)